community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/GeneratedQueryStructure.scala

```diff
@@ -0,0 +1,737 @@
+            Expression.get(handle.self(), ro))),
+        handle.load("e")
+      ))
+    }
+    result
+  }
+
+  val incoming = Expression.get(staticField[Direction, Direction](Direction.INCOMING.name()))
+  val outgoing = Expression.get(staticField[Direction, Direction](Direction.OUTGOING.name()))
+  val both = Expression.get(staticField[Direction, Direction](Direction.BOTH.name()))
+  val newResultRow = Expression.invoke(Expression.newInstance(typeRef[ResultRowImpl]), MethodReference.constructorReference(typeRef[ResultRowImpl]))
+  val newRelationshipDataExtractor = Expression.invoke(Expression.newInstance(typeRef[RelationshipDataExtractor]), MethodReference.constructorReference(typeRef[RelationshipDataExtractor]))
+
+  val CONSTRUCTOR = MethodTemplate.constructor(
+    param[TaskCloser]("closer"),
+    param[Statement]("statement"),
+    param[GraphDatabaseService]("db"),
+    param[ExecutionMode]("executionMode"),
+    param[Supplier[InternalPlanDescription]]("description"),
+    param[QueryExecutionTracer]("tracer"),
+    param[util.Map[String, Object]]("params")).
+    put(self(), typeRef[TaskCloser], "closer", load("closer")).
+    put(self(), typeRef[ReadOperations], "ro", invoke(load("statement"), method[Statement, ReadOperations]("readOperations"))).
+    put(self(), typeRef[GraphDatabaseService], "db", load("db")).
+    put(self(), typeRef[ExecutionMode], "executionMode", load("executionMode")).
+    put(self(), typeRef[Supplier[InternalPlanDescription]], "description", load("description")).
+    put(self(), typeRef[QueryExecutionTracer], "tracer", load("tracer")).
+    put(self(), typeRef[util.Map[String, Object]], "params", load("params")).
+    build()
+
+  val SET_SUCCESSFUL_CLOSEABLE = MethodTemplate.method(typeRef[Unit], "setSuccessfulCloseable",
+    param[SuccessfulCloseable]("closeable")).
+    put(self(), typeRef[SuccessfulCloseable], "closeable", load("closeable")).
+    build()
+  val SUCCESS = MethodTemplate.method(typeRef[Unit], "success").
+    expression(invoke(get(self(), typeRef[SuccessfulCloseable], "closeable"), method[SuccessfulCloseable, Unit]("success"))).
+    build()
+  val CLOSE = MethodTemplate.method(typeRef[Unit], "close").
+    expression(invoke(get(self(), typeRef[SuccessfulCloseable], "closeable"), method[SuccessfulCloseable, Unit]("close"))).
+    build()
+  val EXECUTION_MODE = MethodTemplate.method(typeRef[ExecutionMode], "executionMode").
+    returns(get(self(), typeRef[ExecutionMode], "executionMode")).
+    build()
+  val EXECUTION_PLAN_DESCRIPTION = MethodTemplate.method(typeRef[InternalPlanDescription], "executionPlanDescription").
+    returns(invoke(get(self(), typeRef[Supplier[InternalPlanDescription]], "description"), method[Supplier[InternalPlanDescription], InternalPlanDescription]("get"))).
+    build()
+  val JAVA_COLUMNS = MethodTemplate.method(typeRef[util.List[String]], "javaColumns").
+    returns(get(typeRef[util.List[String]], "COLUMNS")).
+    build()
+}
+
```

Comment:
At some point we should break up this file, it is getting longer by the minute.  Not necessary right now though.